ovs/lib/helpers/decorators.py

```diff
@@ -75,7 +75,33 @@ def new_function(*args, **kwargs):
     return wrap
 
 
-def ensure_single(task_name, extra_task_names=None, mode='DEFAULT', global_timeout=300):
+def ovs_task(**kwargs):
+    """
+    Decorator to execute celery tasks in OVS
+    These tasks can be wrapped additionally in the ensure single decorator
+    """
+    if 'name' not in kwargs:
+        raise ValueError('Name is a mandatory keyword argument for the ovs_task decorator')
+
+    def wrapper(f):
+        """
+        Wrapper function
+        """
+        from ovs.celery_run import celery
+
+        ensure_single_info = kwargs.pop('ensure_single_info', {})
+        if ensure_single_info != {} and os.environ.get('RUNNING_UNITTESTS') != 'True':
+            mode = ensure_single_info.pop('mode', None)
+            if mode is None:
+                raise ValueError('Mode is a mandatory keyword argument for the _ensure_single decorator')
```

Comment:
Why manually raise an error while the python runtime would check this for you and raise the correct expected exception instead?